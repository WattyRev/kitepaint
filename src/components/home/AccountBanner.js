import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { H2, P } from "../../theme";

export const StyleWrapper = styled.div`
  background: ${({ theme }) => theme.colors.silver};
  padding: 16px 24px;
  display: flex;
  justify-content: center;

  .message {
    padding: 16px;
  }
`;

/**
 * Displays a banner with login or account information.
 */
const AccountBanner = ({ isRecognizedUser, children }) => {
  const recognizedUserContent = (
    <div className="message">
      <H2>Welcome Back!</H2>
      <P>Sign in to get back to your saved designs.</P>
    </div>
  );
  const unrecognizedUserContent = (
    <div className="message">
      <H2>Create an Account</H2>
      <P>By creating a free account, you get access to features such as:</P>
      <ul>
        <li>
          <P>Save your designs and modify them later</P>
        </li>
        <li>
          <P>Control the visibility of your designs</P>
        </li>
        <li>
          <P>More features to come!</P>
        </li>
      </ul>
    </div>
  );
  return (
    <StyleWrapper>
      {children}
      {isRecognizedUser ? recognizedUserContent : unrecognizedUserContent}
    </StyleWrapper>
  );
};

AccountBanner.propTypes = {
  /**
   * Indicates if the user is known to have a KitePaint account or not.
   */
  isRecognizedUser: PropTypes.bool,
  /**
   * Children nodes to be displayed to the left of the message.
   */
  children: PropTypes.node.isRequired
};

export default AccountBanner;
