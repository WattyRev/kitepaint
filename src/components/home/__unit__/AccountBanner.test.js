import React from "react";
import { shallow } from "enzyme";
import AccountBanner from "../AccountBanner";

describe("AccountBanner", () => {
  let defaultProps;
  beforeEach(() => {
    defaultProps = {};
  });
  it("renders", () => {
    expect.assertions(1);
    const wrapper = shallow(
      <AccountBanner {...defaultProps}>
        <div>Hello!</div>
      </AccountBanner>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
