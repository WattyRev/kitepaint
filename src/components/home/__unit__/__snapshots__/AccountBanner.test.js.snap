// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountBanner renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccountBanner>
    <div>
      Hello!
    </div>
  </AccountBanner>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div>
          Hello!
        </div>,
        <div
          className="message"
        >
          <ForwardRef>
            Create an Account
          </ForwardRef>
          <P>
            By creating a free account, you get access to features such as:
          </P>
          <ul>
            <li>
              <P>
                Save your designs and modify them later
              </P>
            </li>
            <li>
              <P>
                Control the visibility of your designs
              </P>
            </li>
            <li>
              <P>
                More features to come!
              </P>
            </li>
          </ul>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello!",
        },
        "ref": null,
        "rendered": "Hello!",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ForwardRef>
              Create an Account
            </ForwardRef>,
            <P>
              By creating a free account, you get access to features such as:
            </P>,
            <ul>
              <li>
                <P>
                  Save your designs and modify them later
                </P>
              </li>
              <li>
                <P>
                  Control the visibility of your designs
                </P>
              </li>
              <li>
                <P>
                  More features to come!
                </P>
              </li>
            </ul>,
          ],
          "className": "message",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Create an Account",
            },
            "ref": null,
            "rendered": "Create an Account",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": true,
                "rules": Array [
                  "
  font-size: 1.5em;
",
                ],
              },
              "displayName": "Styled(H2)",
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "By creating a free account, you get access to features such as:",
            },
            "ref": null,
            "rendered": "By creating a free account, you get access to features such as:",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <P>
                    Save your designs and modify them later
                  </P>
                </li>,
                <li>
                  <P>
                    Control the visibility of your designs
                  </P>
                </li>,
                <li>
                  <P>
                    More features to come!
                  </P>
                </li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <P>
                    Save your designs and modify them later
                  </P>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Save your designs and modify them later",
                  },
                  "ref": null,
                  "rendered": "Save your designs and modify them later",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <P>
                    Control the visibility of your designs
                  </P>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Control the visibility of your designs",
                  },
                  "ref": null,
                  "rendered": "Control the visibility of your designs",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <P>
                    More features to come!
                  </P>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "More features to come!",
                  },
                  "ref": null,
                  "rendered": "More features to come!",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-htoDjs",
        "isStatic": false,
        "rules": Array [
          "
  background: ",
          [Function],
          ";
  padding: 16px 24px;
  display: flex;
  justify-content: center;

  .message {
    padding: 16px;
  }
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-htoDjs",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
            Hello!
          </div>,
          <div
            className="message"
          >
            <ForwardRef>
              Create an Account
            </ForwardRef>
            <P>
              By creating a free account, you get access to features such as:
            </P>
            <ul>
              <li>
                <P>
                  Save your designs and modify them later
                </P>
              </li>
              <li>
                <P>
                  Control the visibility of your designs
                </P>
              </li>
              <li>
                <P>
                  More features to come!
                </P>
              </li>
            </ul>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello!",
          },
          "ref": null,
          "rendered": "Hello!",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ForwardRef>
                Create an Account
              </ForwardRef>,
              <P>
                By creating a free account, you get access to features such as:
              </P>,
              <ul>
                <li>
                  <P>
                    Save your designs and modify them later
                  </P>
                </li>
                <li>
                  <P>
                    Control the visibility of your designs
                  </P>
                </li>
                <li>
                  <P>
                    More features to come!
                  </P>
                </li>
              </ul>,
            ],
            "className": "message",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Create an Account",
              },
              "ref": null,
              "rendered": "Create an Account",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": true,
                  "rules": Array [
                    "
  font-size: 1.5em;
",
                  ],
                },
                "displayName": "Styled(H2)",
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "By creating a free account, you get access to features such as:",
              },
              "ref": null,
              "rendered": "By creating a free account, you get access to features such as:",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    <P>
                      Save your designs and modify them later
                    </P>
                  </li>,
                  <li>
                    <P>
                      Control the visibility of your designs
                    </P>
                  </li>,
                  <li>
                    <P>
                      More features to come!
                    </P>
                  </li>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <P>
                      Save your designs and modify them later
                    </P>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Save your designs and modify them later",
                    },
                    "ref": null,
                    "rendered": "Save your designs and modify them later",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <P>
                      Control the visibility of your designs
                    </P>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Control the visibility of your designs",
                    },
                    "ref": null,
                    "rendered": "Control the visibility of your designs",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <P>
                      More features to come!
                    </P>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "More features to come!",
                    },
                    "ref": null,
                    "rendered": "More features to come!",
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "div",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-htoDjs",
          "isStatic": false,
          "rules": Array [
            "
  background: ",
            [Function],
            ";
  padding: 16px 24px;
  display: flex;
  justify-content: center;

  .message {
    padding: 16px;
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-htoDjs",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
